packages/app/src/api.ts

```diff
@@ -220,41 +221,62 @@ export function initializeApp(
  *
  * @param options - `Firebase.AppOptions` to configure the app's services, or a
  *   a `FirebaseApp` instance which contains the `AppOptions` within.
- * @param config - `FirebaseServerApp` configuration.
+ * @param config - Optional `FirebaseServerApp` settings.
  *
  * @returns The initialized `FirebaseServerApp`.
  *
  * @public
  */
 export function initializeServerApp(
   options: FirebaseOptions | FirebaseApp,
-  config: FirebaseServerAppSettings
+  config?: FirebaseServerAppSettings
 ): FirebaseServerApp;
 
+/**
+ * Creates and initializes a {@link @firebase/app#FirebaseServerApp} instance.
+ *
+ * @param config - Optional `FirebaseServerApp` configuration.
+ *
+ * @returns The initialized `FirebaseServerApp`.
+ *
+ * @public
+ */
+export function initializeServerApp(
+  config?: FirebaseServerAppSettings
+): FirebaseServerApp;
 export function initializeServerApp(
-  _options: FirebaseOptions | FirebaseApp,
-  _serverAppConfig: FirebaseServerAppSettings
+  _options?: FirebaseApp | FirebaseServerAppSettings | FirebaseOptions,
+  _serverAppConfig: FirebaseServerAppSettings = {}
 ): FirebaseServerApp {
   if (isBrowser() && !isWebWorker()) {
     // FirebaseServerApp isn't designed to be run in browsers.
     throw ERROR_FACTORY.create(AppError.INVALID_SERVER_APP_ENVIRONMENT);
   }
 
-  if (_serverAppConfig.automaticDataCollectionEnabled === undefined) {
-    _serverAppConfig.automaticDataCollectionEnabled = true;
+  let app: FirebaseApp;
+  let firebaseOptions: FirebaseOptions | undefined;
+  let serverAppSettings: FirebaseServerAppSettings = _serverAppConfig || {};
+
+  if (_options) {
+    if (_isFirebaseApp(_options)) {
+      app = _options;
+      firebaseOptions = app.options;
+    } else if (_isFirebaseServerAppSettings(_options)) {
+      serverAppSettings = _options;
+    } else {
+      firebaseOptions = _options;
+    }
   }
 
-  let appOptions: FirebaseOptions;
-  if (_isFirebaseApp(_options)) {
-    appOptions = _options.options;
-  } else {
-    appOptions = _options;
+  firebaseOptions ||= getDefaultAppConfig();
```

Comment:
I was going to say that it's weird that any default app config found will override any options explicitly passed to `initializeServerApp()` but that's the same behavior as `initializeApp()` so I guess we should stick with it. But I don't know if that's expected behavior for either of them. We should keep it in mind anyway if anyone complains about it on App Hosting or Firebase Studio or anywhere else that has a default environment config.

Reply:
Does it? I thought this would assign the default options only if `firebaseOptions` is undefined.